src/Symfony/Component/Security/Http/Firewall/SimpleFormAuthenticationListener.php

```diff
@@ -107,14 +108,16 @@ protected function attemptAuthentication(Request $request)
             }
         }
 
-        if ($this->options['post_only']) {
-            $username = trim($request->request->get($this->options['username_parameter'], null, true));
-            $password = $request->request->get($this->options['password_parameter'], null, true);
-        } else {
-            $username = trim($request->get($this->options['username_parameter'], null, true));
-            $password = $request->get($this->options['password_parameter'], null, true);
+        $requestBag = $this->options['post_only'] ? $request->request : $request;
+        $username = $requestBag->get($this->options['username_parameter'], null, true);
+        $password = $requestBag->get($this->options['password_parameter'], null, true);
+
+        if (!is_string($username)) {
+            throw new BadRequestHttpException(sprintf('The key "%s" must be a string.', $this->options['username_parameter']));
```

Comment:
This kind of messages in Symfony are usually like:

```php
sprintf('The XXX must be a string, "%s" given.', gettype($XXX))
```

Is it necessary to do the same here?

Reply:
done